File: Example_ObjC/FEKYC_Sample/FEKYC_Sample/ViewController.m
//
//  ViewController.m
//  FEKYC_Sample
//
//  Created by Ragnar on 8/17/20.
//  Copyright © 2020 Ragnar. All rights reserved.
//

#import "ViewController.h"
@import FEKYC;

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
}
- (IBAction)btnStartClicked:(id)sender {
    FEKYCConfig* config = [[FEKYCConfig alloc]initWithApiKey:@"papFhWBwHBV7RvFx7b0STPAZw0xo7kRJ" fullName:@"" orcType:FEKYCOrcTypeVideo orcDocumentType:FEKYCOrcDocumentTypeIdCard isShowResult:false isReturnPhoto:false];
    FEKYC* fe = [[FEKYC alloc] initWithConfig:config];
    
    [fe startFrom:self completion:^(NSDictionary<NSString *,id> * _Nullable result) {
        [[self tvResult] setText:[NSString stringWithFormat:@"%@", result]];
        
        NSDictionary * orcResponse = [result objectForKey:@"orcResponse"];
        NSString* stAddress = [[[[orcResponse objectForKey:@"data"] objectForKey:@"back"] objectForKey:@"data"] objectForKey:@"address"];
        int addressProb = [[[[[orcResponse objectForKey:@"data"] objectForKey:@"back"] objectForKey:@"data"] objectForKey:@"address_prob"] intValue];
        NSLog(@"%@ \n %d", stAddress, addressProb);
    }];
}


@end

File: Example/EKYC_Sample/ViewController.swift
//
//  ViewController.swift
//  EKYC_Sample
//
//  Created by Ragnar on 8/12/20.
//  Copyright © 2020 Ragnar. All rights reserved.
//

import UIKit
import FEKYC
import RadioGroup

class ViewController: UIViewController {

    // MARK: - Properties
    
    // MARK: - Outlet
    @IBOutlet weak var tfFullname: UITextField!
    @IBOutlet var orcDocumentTypeGroup: RadioGroup!
    
    @IBOutlet weak var orcTypeGroup: RadioGroup!
    
    @IBOutlet weak var tvResult: UITextView!
    
    // MARK: - Life cycle
    override func viewDidLoad() {
        super.viewDidLoad()
        
        // Do any additional setup after loading the view.
    
        orcDocumentTypeGroup.titles = ["ID Card", "Driver License", "Passport"]
        orcDocumentTypeGroup.addTarget(self, action: #selector(didSelectOrcDocumentTypeOption(radioGroup:)), for: .valueChanged)
        
        orcTypeGroup.titles = ["Photo", "Liveness"]
        orcTypeGroup.addTarget(self, action: #selector(didSelectOrcTypeOption(radioGroup:)), for: .valueChanged)
    }
    
    // MARK: - Action
    @IBAction func btnStartClicked(_ sender: Any) {
        var orcDocumentType = FEKYCOrcDocumentType.idCard
        var orcType = FEKYCOrcType.photo
        
        switch orcDocumentTypeGroup.selectedIndex {
        case 0:
            orcDocumentType = FEKYCOrcDocumentType.idCard
        case 1:
            orcDocumentType = FEKYCOrcDocumentType.driveLicense
        case 2:
            orcDocumentType = FEKYCOrcDocumentType.passport
        default: break
        }
        
        switch orcTypeGroup.selectedIndex {
        case 0:
            orcType = FEKYCOrcType.photo
        case 1:
            orcType = FEKYCOrcType.liveness
        default: break
        }
        
        let config = FEKYCConfig(apiKey: "papFhWBwHBV7RvFx7b0STPAZw0xo7kRJ", fullName: tfFullname.text!, orcType: orcType, orcDocumentType: orcDocumentType)
        let fekyc = FEKYC(config: config)
        fekyc.start(from: self) { [weak self] result in
            self?.tvResult.text = "\(String(describing: result))"
        }
    }
    
    
    @objc func didSelectOrcDocumentTypeOption(radioGroup: RadioGroup) {
        print(radioGroup.titles[radioGroup.selectedIndex] ?? "")
    }

    @objc func didSelectOrcTypeOption(radioGroup: RadioGroup) {
        print(radioGroup.titles[radioGroup.selectedIndex] ?? "")
    }
}

